### DISCUSSION TITLE
URLs pasted into editor are converted to HTML tags on some characters.

### DISCUSSION THREAD

**Discussion by mpemburn** [ueberdosis/tiptap#1476] [General Questions]
The web app I work on uses Tiptap for some of our fields.  Today, I was given the task of figuring out why some URLs break when they are pasted in.  For example, this:

https://www.amazon.com/s?k=1-3%2F8+masonry+drill+bit&gclid=CjwKCAjwxuuCBhATEiwAIIIz0fvajzsZFUUsV_hYpmT_81yl-I8lDCFO2m0ja0crmO_jnZF3lRQavxoC7gUQAvD_BwE&hvadid=439005554383&hvdev=c&hvlocphy=9007821&hvnetw=g&hvqmt=e&hvrand=14248236890966909551&hvtargid=kwd-915190254183&hydadcr=28597_10166504&tag=googhydr-20&ref=pd_sl_3y3dp0ia3t_e

...gets turned into this:

<p><a href="https://www.amazon.com/s?k=1-3%2F8+masonry+drill+bit&amp;gclid=CjwKCAjwxuuCBhATEiwAIIIz0fvajzsZFUUsV" rel="noopener noreferrer nofollow">https://www.amazon.com/s?k=1-3%2F8+masonry+drill+bit&amp;gclid=CjwKCAjwxuuCBhATEiwAIIIz0fvajzsZFUUsV</a><em>hYpmT</em>81yl-I8lDCFO2m0ja0crmO<em>jnZF3lRQavxoC7gUQAvD</em>BwE&amp;hvadid=439005554383&amp;hvdev=c&amp;hvlocphy=9007821&amp;hvnetw=g&amp;hvqmt=e&amp;hvrand=14248236890966909551&amp;hvtargid=kwd-915190254183&amp;hydadcr=28597<em>10166504&amp;tag=googhydr-20&amp;ref=pd</em>sl<em>3y3dp0ia3t</em>e</p>

What appears to be happening is that any text that is surrounded by underscores is seen as italics.  Is there any way around this?

Thanks in advance,

Mark Pemburn
Trak Software

**Comment by philippkuehn** [comment]
are you using tiptap v1 or v2?

**Reply by mpemburn** [reply]
Looks like we're on v1.32.2

**Reply by philippkuehn** [reply]
Yeah that was a bug in v1 and is fixed in v2 as you can test here: https://www.tiptap.dev/api/marks/link/#link

**Reply by mpemburn** [reply]
Wonderful—thank you!  I don't think I can convince the CTO to use a beta version, however.  Do you have a projected date for general release?

**Reply by hanspagel** [reply]
GitLab is using it in production on thousands of installations:

https://gitlab.com/gitlab-org/gitlab/-/blob/master/package.json#L65

It’s safe to use. Just don’t be to upset if we’ll rename a command or something else before the stable release.

**Reply by mpemburn** [reply]
Hi folks,

I got the go-ahead to use the beta version, and I tried to add it to my project without success.  We are using yarn, so I tried:

`yarn add tiptap@2.0.0-beta.75`

This gave me the error:
Couldn't find any versions for "tiptap" that matches "2.0.0-beta.75"—and a list of versions that showed 1.32.2 as the latest.

Can you tell me the correct way to add this?  Thanks!

**Reply by philippkuehn** [reply]
for v2 we are using scoped packages (@tiptap).

here is a migration guide:

https://www.tiptap.dev/overview/upgrade-guide
